src/main/java/org/apache/bcel/classfile/JavaClass.java

```diff
@@ -906,4 +911,28 @@ public String toString() {
         }
         return buf.toString();
     }
+
+    /**
+     * Tests whether this class was declared as a record
+     *
+     * @return true if a record attribute is present, false otherwise.
+     * @since 6.9.0
+     */
+    public boolean isRecord() {
+        computeIsRecord();
+        return this.isRecord;
+    }
+
+    private void computeIsRecord() {
+        if (computedRecord) {
+            return;
+        }
+        for (final Attribute attribute : this.attributes) {
+            if (attribute instanceof Record) {
+                isRecord = true;
+                break;
+            }
+        }
+        this.computedRecord = false;
```

Comment:
shouldn't this be set to true?

Reply:
fixed in https://github.com/apache/commons-bcel/pull/290/commits/e3c2f2eff322f998fd97416becb60c68d6a3ba48